planner/core/stats.go

```diff
@@ -177,18 +178,14 @@ func (ds *DataSource) DeriveStats(childStats []*property.StatsInfo) (*property.S
 		}
 	}
 	// Consider the IndexMergePath. Now, we just generate `IndexMergePath` in DNF case.
-	if len(ds.pushedDownConds) > 0 && len(ds.possibleAccessPaths) > 1 && ds.ctx.GetSessionVars().GetEnableIndexMerge() {
-		needConsiderIndexMerge := true
-		for i := 1; i < len(ds.possibleAccessPaths); i++ {
-			if len(ds.possibleAccessPaths[i].accessConds) != 0 {
-				needConsiderIndexMerge = false
-				break
-			}
-		}
-		if needConsiderIndexMerge {
```

Comment:
Why is this check removed?

Reply:
I consider we can just use the` EnableIndexMerge` to control the behavior. However, we can eliminate some operations with it.

Reply:
`select * from t where (a < 1 or b < 1) and c < 100000;` We would better remove it. `c < 1` is in `accessConds`. However, IndexMerge on `(a < 1 or b < 1)`may be better according to the cost.

Reply:
I prefer to keep this check at least for now, until the index merge is fully tested, otherwise it may cause performance regression or fluctuation if the cost estimation is not that accurate (plan changed from existing index scan to index merge). We can add a TODO here to inform removing it later.